src/java.base/share/classes/java/nio/MappedByteBuffer.java

```diff
@@ -353,4 +366,41 @@ public final MappedByteBuffer rewind() {
         super.rewind();
         return this;
     }
+
+    /**
+     * {@inheritDoc}
+     *
+     * <p> Reading bytes into physical memory by invoking {@code load()} on the
+     * returned buffer, or writing bytes to the storage device by invoking
+     * {@code force()} on the returned buffer, will only act on the sub-range
+     * of this buffer that the returned buffer represents, namely
+     * {@code [position(),limit()]}.
```

Comment:
Is the closing bracket on this interval meant to be ']' or ')'? i.e. is the interval inclusive of exclusive of the value at the index returned by limit()?

Reply:
The no-arg force method used to write back the entire buffer content [0, capacity]. It has been changed to use the limit, meaning [0, limit]. I don't recall this being an intentional change. I'm asking now because the change in this PR will specify the new behavior rather than the long standing behavior.

Reply:
@AlanBateman Hmm, I think that's a different issue to my comment.

I asked my question above because the correction Brian made when documenting slice(int index, int length) employed a half-closed interval with an exclusive right bound for the interval

    [index,index+length)

and I merely felt that the same should apply here in the documentation of slice()

  [position(),limit())

I'm not sure I am in a position to provide the right answer to your question but I'll note that in the same pedantic vein by the same token the intervals you mention should be similarly semi-closed

    [0, capacity)

and

    [0, limit)

The byte at index limit/capacity is not necessarily flushed (modulo page roundings).

As to what might be a correct answer:

> The no-arg force method used to write back the entire buffer content [0, capacity]. It has been changed to use the limit, meaning [0, limit]. I don't recall this being an intentional change.

I believe this was done intentionally when I introduced the force(int,int) variant and re-implemented the noargs variant to indirect to it. I cannot recall why it changed from capacity to limit though.

Reply:
@adinn My comment wasn't to you, it's a concern that the no-arg force method has changed and the new behavior is proposed to be specified by this PR. I need to re-read the mails as it may be that this will need to be restored to use the capacity.

Reply:
The right hand limit should in all cases be closed so the current no-arg doc is wrong.

As for `force()` changing from `[0,capacity())` to `[0,limit())`, I don't know exactly why that happened or when but it definitely changed since Java 8.

Reply:
np, and I assume we'll continue this PR assuming JDK-8263742 will be integrated.